packages/queries/src/schema/getSelectableColumns.ts

```diff
@@ -25,18 +33,58 @@ type SelectableColumnsType = Record<string, true>;
  */
 
 export function getSelectableColumns<DB>(
-    schema: DbSchema<DB>,
+    schema: Schema<DB>,
     table: Table<DB>,
 ): SelectableColumnsType {
     const select: SelectableColumnsType = {};
 
     const tableDef = getTableDef<DB>(schema, table);
 
-    for (const [columnName, columnDef] of getColumnDefs<DB>(tableDef)) {
+    for (const [columnName, columnDef] of getColumnNamesAndDefs<DB>(tableDef)) {
         if (isSelectableColumn(columnDef)) {
             select[columnName] = true;
         }
     }
 
     return select;
 }
+
+export function getSelectableColumnsFromGenericDbSchema(
```

Comment:
it's now a bit weird that getSelectableColumns.ts is storing all these different functions.

My preference is to have one logical unit per file. This is not always possible, but I think it is in this case.